sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/V1Writes.scala

```diff
@@ -107,7 +107,7 @@ object V1Writes extends Rule[LogicalPlan] with SQLConfHelper {
     if (orderingMatched) {
       empty2NullPlan
     } else {
-      Sort(requiredOrdering, global = false, empty2NullPlan)
+      Clustering(requiredOrdering, empty2NullPlan)
```

Comment:
Can we add a config to protect this switch? In case any bug happens.

Reply:
Addressed.